contracts/wawaka/interpreter-test/interpreter-test.cpp

```diff
@@ -125,6 +126,44 @@ bool ecdsa_test(const Message& msg, const Environment& env, Response& rsp)
     return rsp.value(v, false);
 }
 
+// XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+// NAME: extended_ecdsa_test
+// XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+bool extended_ecdsa_test(const Message& msg, const Environment& env, Response& rsp)
+{
+    const std::string message_string(msg.get_string("message"));
+    const ww::types::ByteArray message(message_string.begin(), message_string.end());
+
+    // ---------- get the keys we need ----------
+    ww::types::ByteArray bytes(48);
+    ASSERT_SUCCESS(
+        rsp, ww::crypto::random_identifier(bytes),
```

Comment:
btw: this doesn't create a completely uniform key but i guess that's ok here ...

Reply:
for the purposes of the test. i'm not worried.
this test could definitely be extended to be more robust. i'm trusting the tests in the common library for "correctness" and this is really about verifying the connection between the contract and the common library code.